[refarch-gateway] Bug/refarch gateway fix hazelcast

simonhir opened the pull request:
**Description**

- Add missing port for Hazelcast
- Add shortcut env var `HAZELCAST_JAVA_OPTS_APPEND` which can be used in `JAVA_OPTS_APPEND`
- Bump version to 1.3.0

**Reference**

Issues https://github.com/it-at-m/refarch/issues/179

simonhir committed:
:bug: refarch-gateway add missing hazelcast port

simonhir committed:
:sparkles: refarch-gateway add hazelcast java opts shortcut env var

simonhir committed:
:pushpin: refarch-gateway use version 1.3.0

simonhir committed:
:recycle: refarch-gateway service use hazelcast port name

simonhir committed:
:bug: refarch-gateway deployment fix additional env empty

hupling commented on charts/refarch-gateway/templates/deployment.yaml:
```diff
@@ -50,7 +55,13 @@ spec:
+            {{- if .Values.hazelcast.enabled }}
+            - name: HAZELCAST_JAVA_OPTS_APPEND
+              value: "--add-modules java.se --add-exports java.base/jdk.internal.ref=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.management/sun.management=ALL-UNNAMED --add-opens jdk.management/com.ibm.lang.management.internal=ALL-UNNAMED --add-opens jdk.management/com.sun.manageme ...
+            {{- end }}
```
@ejcsid meinte man soll irgendwie .Values.env mit den Standardwerten machen und Values.envAppend für zusätzliche Sachen, die die Leute selbstständig definieren.

Hier fehlt noch die Zeitzone und GC_MAX_METASPACE_SIZE: "200"

hupling commented:
Die Frage ist, ob man hier nicht auch einen Service-Account generieren lässt mit den entsprechenden Rechten und nicht den Default-Account hernimmt.

hupling commented:
https://docs.hazelcast.com/hazelcast/5.5/kubernetes/kubernetes-auto-discovery#hazelcast-configuration

> If you don’t specify service-name and fall back to a namespace only discovery, all pods in the namespace must be Hazelcast member pods; other pods might block the member discovery process of the Hazelcast member pods. 

Also den ServiceName würde ich auch angeben. Diesen würde ich aber dynmisch machen. Dann brauchst du wahrscheinlich die tpl funktionion. https://github.com/it-at-m/helm-charts/pull/57/files#discussion_r1791442192

hupling commented:
> namespace: Kubernetes Namespace where Hazelcast is running; if not specified, the value is taken from the environment variables KUBERNETES_NAMESPACE or OPENSHIFT_BUILD_NAMESPACE. If those are not set, the namespace of the Pod will be used (retrieved from /var/run/secrets/kubernetes.io/serviceaccount/namespace).

Mit dem Namespace sollten wir kein Problem mehr haben, da wir das Image nicht mehr in Openshift bauen.

simonhir committed:
:sparkles: refarch-gateway deployment preset hazelcast service name

simonhir committed:
:sparkles: refarch-gateway add hazelcast role and binding

simonhir commented:
@hupling 

- Added custom role and role binding for service-account
- Added `REFARCH_HAZELCAST_SERVICENAME`
- Removed `REFARCH_HAZELCAST_NAMESPACENAME` in https://github.com/it-at-m/refarch/pull/206

hupling commented on charts/refarch-gateway/templates/deployment.yaml:
```diff
@@ -50,7 +55,15 @@ spec:
             {{- toYaml .Values.resources | nindent 12 }}
           env:
+            {{- if gt (len .Values.env) 0 }}
             {{- toYaml .Values.env | nindent 12 }}
```
```suggestion
            {{- tpl (toYaml .Values.env) . | nindent 12 }}
```

hupling commented on charts/refarch-gateway/templates/deployment.yaml:
```diff
@@ -50,7 +55,15 @@ spec:
           env:
+            {{- if gt (len .Values.env) 0 }}
             {{- toYaml .Values.env | nindent 12 }}
+            {{- end}}
```
```suggestion
            {{- end}}
            {{- if gt (len .Values.envAppend) 0 }}
            {{- tpl (toYaml .Values.envAppend) . | nindent 12 }}
            {{- end}}
```

hupling committed:
default values

hupling committed:
fix lint

hupling committed:
tpl

hupling committed:
add extra env

hupling committed:
fix lint

hupling committed:
Merge branch 'bug/refarch-gateway-fix-hazelcast' of https://github.com/it-at-m/helm-charts into bug/refarch-gateway-fix-hazelcast

simonhir commented:
@hupling 
In my opinion the value refactoring should be done in a separate pr. Has nothing to do with this Hazelcast change

hupling reviewed (approved):
LGTM 

Ich war jetzt noch so frei und habe https://github.com/it-at-m/helm-charts/pull/43 das eingebaut

hupling commented:
> @hupling In my opinion the value refactoring should be done in a separate pr. Has nothing to do with this Hazelcast change

Wie mans sieht, ich hätte halt das ganze Zeug in die values.yml reingemacht und nicht in das Template.

```yml
            {{- if .Values.hazelcast.enabled }}
            - name: HAZELCAST_JAVA_OPTS_APPEND
              value: "--add-modules java.se --add-exports java.base/jdk.internal.ref=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.management/sun.management=ALL-UNNAMED --add-opens jdk.management/com.ibm.lang.management.internal=ALL-UNNAMED --add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED"
            - name: REFARCH_HAZELCAST_SERVICENAME
              value: {{ include "refarch-gateway.fullname" . }}
            {{- end }}
````

simonhir committed:
:bookmark: refarch-gateway version 1.3.1

simonhir commented on charts/refarch-gateway/a.txt:
@hupling
Pushed by accident?

simonhir replied:
Was removed

devtobi commented on charts/refarch-gateway/templates/service.yaml:
```diff
@@ -11,5 +11,11 @@ spec:
+    - name: hazelcast
+      protocol: TCP
+      port: 5701
+      targetPort: hazelcast
```
Is this a reference to the port name defined in the service?

simonhir replied:
Yes, in the deployment

hupling committed:
Delete charts/refarch-gateway/a.txt

devtobi reviewed (approved):
LGTM
